Service/Form/Submission/FormSubmissionService.php

```diff
@@ -91,4 +91,24 @@ public function submit(FormSubmissionRequest $submitRequest): array
 
         return ['submission_id' => $formSubmission->getId()];
     }
+
+    public function removeExpiredSubmissions(): int
+    {
+        $removedCount = 0;
+
+        foreach ($this->repository->findAll() as $submission) {
+            $deadline = $submission->getDeadlineDate();
```

Comment:
The repository should implement a function like getAllOutdatedSubmission?

Reply:
Agree

Reply:
indeed, this function should call respository->findAllOutdatedSubmissions($expirationDate).

And then delete everything that is returned.

Reply:
@dameert Do we need the expirationDate argument? We just need to find those who are in the past right?